src/ws/remotectl-certificate.c

```diff
@@ -171,21 +171,30 @@ ensure_certificate (const gchar *user,
       goto out;
     }
 
-  /* auto-renew our expired auto-created certificates */
-  /* must specify ca due to https://gitlab.gnome.org/GNOME/glib-networking/-/issues/139 */
-  if (check_expired && g_str_has_suffix (path, "/0-self-signed.cert") &&
-      g_tls_certificate_verify (certificate, NULL, certificate) & G_TLS_CERTIFICATE_EXPIRED)
+  /* only manage/change our own certificate */
+  if (g_str_has_suffix (path, "/0-self-signed.cert"))
     {
-      int r;
-      g_message ("Automatically created certificate %s is expired, renewing it", path);
-      r = g_unlink (path);
-      if (r == 0)
-        ensure_certificate (user, group, selinux, FALSE);  /* avoid infinite recursion */
-      else
-        g_warning ("Failed to remove %s: %m", path);
-    }
+      /* auto-renew our expired auto-created certificates */
+      /* must specify ca due to https://gitlab.gnome.org/GNOME/glib-networking/-/issues/139 */
+      if (check_expired &&
+          g_tls_certificate_verify (certificate, NULL, certificate) & G_TLS_CERTIFICATE_EXPIRED)
+        {
+          int r;
+          g_message ("Automatically created certificate %s is expired, renewing it", path);
+          r = g_unlink (path);
+          if (r == 0)
+            ensure_certificate (user, group, selinux, FALSE);  /* avoid infinite recursion */
+          else
+            g_warning ("Failed to remove %s: %m", path);
+        }
 
-  ret = set_cert_attributes (path, user, group, selinux);
+      ret = set_cert_attributes (path, user, group, selinux);
+    }
+  else
+    {
```

Comment:
This function is *begging* to be rewritten with autoptr, inversion of the "not us?" logic to be an early return, and avoiding the recursion...

Reply:
Added a new commit for g_auto. I had to change the structure a bit, the failed [realms test](https://logs.cockpit-project.org/logs/pull-15321-20210212-093531-65cde94f-fedora-33/log.html#289-2) pointed out that we still need to adjust the permissions for 10-ipa as well.